Source/WebCore/style/StyleScope.h

```diff
@@ -267,6 +272,7 @@ class Scope final : public CanMakeWeakPtr<Scope>, public CanMakeCheckedPtr<Scope
         bool operator==(const AnchorPosition&) const = default;
     };
     SingleThreadWeakHashMap<const RenderBoxModelObject, AnchorPosition> m_anchorPositionsOnLastUpdate;
+    HashMap<AnchorPositionedKey, size_t> m_lastSuccessfulPositionOptionIndexes;
```

Comment:
AnchorPositionedKey refs Element and I think so far it is only used for transient data structures. I'm not entirely sure this patch guarantees cleanup in all case (when elements are removed from there tree etc)? Otherwise we'd leak Elements.

Reply:
The intention here is to have a unique way to identify an element + optional pseudo-element. `AnchorPositionedKey` is the most straightforward way, but we can do better without ref-counting the `Element`. Maybe `Element::nodeIdentifier()` + `PseudoElementIdentifier`?

Reply:
forgetLastSuccessfulPositionOptionIndex is called when position try box is unregistered so maybe that is sufficient to ensure cleanup. Maybe you can put `ASSERT(m_lastSuccessfulPositionOptionIndexes.isEmpty())` to Style::Scope destructor to prove that?